npm/src/upload/index.ts

```diff
@@ -0,0 +1,76 @@
+import { TdrFile } from "@nationalarchives/file-information"
+import { ClientFileProcessing } from "../clientprocessing"
+
+interface HTMLInputTarget extends EventTarget {
+  files?: InputElement
+}
+
+interface InputElement {
+  files?: TdrFile[]
+}
+
+export class UploadFiles {
+  clientFileProcessing: ClientFileProcessing
+
+  constructor(clientFileProcessing: ClientFileProcessing) {
+    this.clientFileProcessing = clientFileProcessing
+  }
+
+  upload(): void {
+    const uploadForm: HTMLFormElement | null = document.querySelector(
+      "#file-upload-form"
+    )
+
+    if (uploadForm) {
+      uploadForm.addEventListener("submit", ev => {
+        ev.preventDefault()
+        const consignmentId: number = this.retrieveConsignmentId()
+
+        if (!consignmentId) {
+          throw Error("No consignment provided")
+        }
+
+        const target: HTMLInputTarget | null = ev.currentTarget
+
+        try {
+          const files: TdrFile[] = target!.files!.files!
+
+          this.generateFileDetails(consignmentId, files.length).then(r => {
+            this.uploadClientFileMetadata(r, files)
+          })
+        } catch (e) {
+          //For now console log errors
+          console.log("Upload failed: " + e.message)
+        }
+      })
+    }
+  }
+
+  retrieveConsignmentId(): number {
+    const pathName: string = window.location.pathname
```

Comment:
I think this will work, but it feels wrong for the JavaScript to be dependent on the URL (except for SPA routing). I can't quite put my finger on why - it might be because we want to get this value from some kind of data store, and we can do that in the HTML without depending on the path. The path is also a bit more fragile: we might add new parts to the path, and we might eventually add the upload form to different pages (e.g. to let users upload files to an existing consignment, or to replace a file, which would have a different URL).

A more idiomatic way to do this might be to add the consignment ID as a [data attribute](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes) in one of the HTML elements - perhaps the form?

```
<form id="file-upload-form" data-consignment-id="the-consignment-uuid">
```

You can then 

There might be an even better way, though! My knowledge of JavaScript best practices is a bit rusty. 🙂